pkg/steps/multi_stage.go

```diff
@@ -172,14 +176,29 @@ func (s *multiStageTestStep) run(ctx context.Context) error {
 		return fmt.Errorf("failed to create RBAC objects: %w", err)
 	}
 	var errs []error
+	observers, err := s.generateObservers(s.observers)
+	if err != nil {
+		// if we can't even generate the Pods there's no reason to run the job
+		return err
+	}
+	observerContext, cancel := context.WithCancel(ctx)
+	observerDone := make(chan interface{})
+	go s.runObservers(observerContext, observers, observerDone)
+
 	if err := s.runSteps(ctx, s.pre, env, true, false); err != nil {
 		errs = append(errs, fmt.Errorf("%q pre steps failed: %w", s.name, err))
 	} else if err := s.runSteps(ctx, s.test, env, true, len(errs) != 0); err != nil {
 		errs = append(errs, fmt.Errorf("%q test steps failed: %w", s.name, err))
 	}
+	cancel() // signal to observers that we're tearing down
 	if err := s.runSteps(context.Background(), s.post, env, false, len(errs) != 0); err != nil {
 		errs = append(errs, fmt.Errorf("%q post steps failed: %w", s.name, err))
 	}
+	// we need to wait for the observers to finish so we get their jUnit
```

Comment:
I don't see it in the E2E test?

For regular steps, that is done by https://github.com/openshift/ci-tools/blob/4df23e041313f8b32bbcdb0f5967017d66ca72e9/pkg/steps/multi_stage.go#L624